addons/modules/modules/ambientAnimals/functions/fnc_module.sqf

```diff
@@ -0,0 +1,49 @@
+#include "script_component.hpp"
+#define EXEC_MODULE_NAME GVAR(ambientAnimals)
+/*
+ * Author: Krzyciu, SzwedzikPL
+ * Ambient animals module
+ */
+params ["_mode","_input"];
+
+// Exit if module executed inside editor, not on server or not in init mode
+if (is3DEN || !(isServer) || !(_mode isEqualTo "init")) exitWith {};
+_input params [
+  ["_logic", objNull, [objNull]],
+  ["_isActivated", false, [true]],
+  ["_isCuratorPlaced", false, [true]]
+];
+// Exit if module is null, not local or placed by zeus (should not happen)
+if (isNull _logic || !(local _logic) || _isCuratorPlaced) exitWith {};
+
+if (CBA_missionTime < 30) then {
```

Comment:
I know the reason why i used it in genSoldiers - compositions. Why it is used here?

Reply:
Tested it on dedicated server, if more than 5 modules with max amount of animals were used at mission start, then animals count didn't match - 1 or 2 were skipped in each generator. Sleep resolved this issue.